Posit Infrastructure/FAQs.md

```diff
@@ -89,6 +90,30 @@ https://pwb.publichealthscotland.org/s/bd664cfc78a31605322d7/?launcher=1
 
 The long string of letters and numbers (in this case <code>bd664cfc78a31605322d7</code>) is the session ID.
 
+
+#### How can I see how much memory my RStudio session is using in Posit Workbench?
+##### 1. Using code
+For the most accurate results, you can run the line of code below. This will return the amount of memory (in megabytes) currently used by your session to the environment as an object called 'memory_used_in_mb':
+
+```{r}
+memory_used_in_mb <- as.numeric(system2("/usr/bin/cat", "/sys/fs/cgroup/memory.current", stdout = TRUE)) / (1000^2)
```

Comment:
Slightly simpler but should work the same as `cat` is a built-in command
```suggestion
memory_used_in_mb <- as.numeric(system2("cat", "/sys/fs/cgroup/memory.current", stdout = TRUE)) / (1000^2)
```

Reply:
Thanks James! Have updated (and committed) the other change (1000 -> 1024) but then it wouldn't let me add this suggestion and says it's outdated. I'll make this change separately and commit it.

Reply:
Great, and yeah sorry, could have done a better job with the comments / suggestions 😅

Reply:
Not sure if 1024 is correct here though? I have the following (which uses 1million bytes = 1 Mb) print the session memory limit when I start up and it's always the same as the number used to start the session.

`as.numeric(system("cat /sys/fs/cgroup/memory.max", intern = TRUE))/1e6`

Reply:
Struggling to find a conclusive reference... (https://unix.stackexchange.com/a/621576/116713 ?) but all that shows is that to get Megabytes you should divide by 1024^2.

Bob's point is interesting as what's really important is what someone will use the number for i.e. what they might compare it to...

Reply:
Well bytes are always unambiguous ... so if `/sys/fs/cgroup/memory.max` is in bytes, then dividing by 1000^2 is definitely megabytes, and dividing by 1024^2 is mebibytes (MiB). People will mix and match but I think that's the accepted definition these days (unless you're trying to sell a hard drive in which case you just pick whatever makes the biggest round number).

Reply:
> Slightly simpler but should work the same as `cat` is a built-in command

There's a (small) possibility that the path to `cat` is not in the user's path, so I'd suggest leaving in the full path here.

Reply:
> Well bytes are always unambiguous ... so if `/sys/fs/cgroup/memory.max` is in bytes, then dividing by 1000^2 is definitely megabytes, and dividing by 1024^2 is mebibytes (MiB). People will mix and match but I think that's the accepted definition these days (unless you're trying to sell a hard drive in which case you just pick whatever makes the biggest round number).

The calculations should definitely be based on 1000^2 as this is how memory allocations and usage are calculated in Kubernetes, hence why when @jakeybob divides by 1e6, he gets the exact amount of memory he requested for his session.

Reply:
> > Slightly simpler but should work the same as `cat` is a built-in command
> 
> There's a (small) possibility that the path to `cat` is not in the user's path, so I'd suggest leaving in the full path here.

I had assumed that `cat` was built-in, but apparently not: 
![image](https://github.com/user-attachments/assets/0c495baa-4f7e-4c71-a334-d484690de928) (https://unix.stackexchange.com/questions/208615/is-cat-a-shell-built-in-or-an-external-program)

I'd still leave it as just `cat` since this possibility seems very remote, and would surely indicate a wider / different issue 😆. Particularly since this is in the FAQ, I would er towards simpler wherever possible.